src/test/java/hudson/plugins/ec2/EC2RetentionStrategyTest.java

```diff
@@ -471,4 +472,11 @@ public void testRetentionStopsAfterActiveRangeEnds() throws Exception {
         assertEquals(1, computers.size());
         assertEquals(1, AmazonEC2FactoryMockImpl.instances.size());
     }
+
+    private static void checkRetentionStrategy(EC2RetentionStrategy rs, EC2Computer c) throws InterruptedException {
+        rs.check(c);
+        EC2AbstractSlave node = c.getNode();
+        while (node.terminateScheduled)
+            Thread.sleep(TimeUnit.SECONDS.toMillis(1));
```

Comment:
At first, I was going to suggest mocking the `Clock` instance in the test to advance time, but then I realized this is a spinlock. To be a more reliable test that depends less on the passage of time (i.e., stateful), use of [CountDownLatch](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CountDownLatch.html) which counts down at the end of the `Runnable` submitted would be more appropriate.

Reply:
Hmm, how do you propose that to work. Since the call happens inside terminate() I don't see a way of introducing it in tests.

Reply:
If you want to make it repeatable, there's a CyclicBarrier class you can use to reset the counter each time. Basically, you count down on terminate, and in the unit test, you use await() on the latch or barrier.

Reply:
As for testability, it'd be ok to expose as package-private for unit tests (I think there might be a `@VisibleForTesting` annotation or something similar you can use).